modules/icgc-argo-workflows/payload/splicejunction/resources/usr/bin/main.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+ Copyright (c) 2024, Ontario Institute for Cancer Research (OICR).
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published
+ by the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+ Author: Guanqiao Feng <gfeng@oicr.on.ca>
+ """
+
+import os
+import sys
+import json
+import sys
+import argparse
+import json
+import hashlib
+import uuid
+from datetime import date
+import copy
+import yaml
+
+
+def calculate_size(file_path):
+    return os.stat(file_path).st_size
+
+
+def calculate_md5(file_path):
+    md5 = hashlib.md5()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(1024 * 1024), b''):
+            md5.update(chunk)
+    return md5.hexdigest()
+
+
+def rename_file(f, payload, rg_count, sample_info, date_str):
+    experimental_strategy = payload['experiment']['experimental_strategy'].lower()
+
+    if f.endswith('.txt'):
+        file_ext = 'txt'
+    else:
+        sys.exit('Error: unknown aligned seq extention: %s' % f)
+
+    new_name = "%s.%s.%s.%s.%s.%s.%s" % (
+        payload['studyId'],
+        sample_info[0]['donor']['donorId'],
+        sample_info[0]['sampleId'],
+        experimental_strategy,
+        date_str,
+        'splice-junctions',
+        file_ext
+    )
```

Comment:
Likewise, can we try to make the filename pattern to be back compatible with the existing files.
Apparently, aligner info is missing from the file name, that could potentially generate the same file names from star and hisat2.